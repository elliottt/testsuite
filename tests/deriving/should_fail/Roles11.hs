{-# LANGUAGE GADTs, RoleAnnotations #-}

module Roles11 where

data T2 a@R where
  K2 :: T2 Int

