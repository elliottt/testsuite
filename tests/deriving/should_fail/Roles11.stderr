
Roles11.hs:5:1:
    Role mismatch on variable a:
      Annotation says Representational but role Nominal is required
    In the data declaration for ‛T2’
